### PR TITLE
Fixes v2 examples serialization in a response object

### LINKED ISSUE #1430: Examples are missing after converting V3 swagger to V2
**Bug:** 
we need to convert V3 swagger to V2 and transform the examples, but examples part in request/responses/components are missing after converting V3 swagger to V2.
**v3 swagger file:** https://github.com/linkedin-developers/job-posting-development-tools/blob/master/OpenAPI-specification/job_posting_v2.yaml

**To Reproduce**
Steps to reproduce the current behavior:
1. run project Microsoft.OpenApi.Workbench
2. copy the content of v3 swagger file and paste it to "input content" area
3. choose JSON for Format, choose V2.0 for Version and click button "Convert" 

**Expected behavior**
examples should display in V2 swagger.

**Screenshots/Code Snippets**
**stack trace:**
>	Microsoft.OpenApi.dll!Microsoft.OpenApi.Models.OpenApiDocument.SerializeAsV2(Microsoft.OpenApi.Writers.IOpenApiWriter writer) Line 143	C#
 	Microsoft.OpenApi.dll!Microsoft.OpenApi.Extensions.OpenApiSerializableExtensions.Serialize<Microsoft.OpenApi.Models.OpenApiDocument>(Microsoft.OpenApi.Models.OpenApiDocument element, Microsoft.OpenApi.Writers.IOpenApiWriter writer, Microsoft.OpenApi.OpenApiSpecVersion specVersion) Line 125	C#
 	Microsoft.OpenApi.dll!Microsoft.OpenApi.Extensions.OpenApiSerializableExtensions.Serialize<Microsoft.OpenApi.Models.OpenApiDocument>(Microsoft.OpenApi.Models.OpenApiDocument element, System.IO.Stream stream, Microsoft.OpenApi.OpenApiSpecVersion specVersion, Microsoft.OpenApi.OpenApiFormat format, Microsoft.OpenApi.Writers.OpenApiWriterSettings settings) Line 94	C#
 	Microsoft.OpenApi.Workbench.dll!Microsoft.OpenApi.Workbench.MainModel.WriteContents(Microsoft.OpenApi.Models.OpenApiDocument document) Line 312	C#
 	Microsoft.OpenApi.Workbench.dll!Microsoft.OpenApi.Workbench.MainModel.ParseDocument() Line 279	C#
 	Microsoft.OpenApi.Workbench.dll!Microsoft.OpenApi.Workbench.MainWindow.Button_Click(object sender, System.Windows.RoutedEventArgs e) Line 26	C#

**screenshot:**
image

image

### PR DESCRIPTION
**Pull request by MaggieKimani1**
- Retains examples present in a media type object by adding serialization logic of the examples map to `x-examples` during down-casting from v3 to v2
- Adds logic to cater for the round trip i.e.  serializing `x-examples` in a v2 document to examples in the resulting v3 document during upcasting
- Fixes #1430 

### Snippets
``` yaml
openapi: 3.0.1
info:
  title: Pet Store API
  version: 1.0.0
paths:
  /pets:
    get:
      summary: Get all pets
      responses:
        '200':
          description: Successful response
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    name:
                      type: string
                    age:
                      type: integer
              examples:
                example1:
                  summary: Example - List of Pets
                  value:
                    - name: "Buddy"
                      age: 2
                    - name: "Whiskers"
                      age: 1
               example2:
                  summary: Example - Playful Cat
                  value:
                    name: "Whiskers"
                    age: 1
```
Results to:
``` yaml
swagger: '2.0'
info:
  title: Pet Store API
  version: 1.0.0
paths:
  /pets:
    get:
      summary: Get all pets
      produces:
        - application/json
      responses:
        '200':
          description: Successful response
          schema:
            type: array
            items:
              type: object
              properties:
                name:
                  type: string
                age:
                  type: integer
          x-examples:
            example1:
              summary: Example - List of Pets
              value:
                - name: "Buddy"
                  age: 2
                - name: "Whiskers"
                  age: 1
           example2:
              summary: Example - Playful Cat
              value:
                name: "Whiskers"
                age: 1
```
And the reverse is also true.

### CODE REVIEW

File: src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs [hunk 1]
```diff
@@ -38,7 +42,7 @@ internal static partial class OpenApiV2Deserializer
         private static readonly PatternFieldMap<OpenApiResponse> _responsePatternFields =
             new()
             {
-                {s => s.StartsWith("x-"), (o, p, n) => o.AddExtension(p, LoadExtension(p, n))}
+                {s => s.StartsWith("x-") && !s.Equals("x-examples"), (o, p, n) => o.AddExtension(p, LoadExtension(p, n))}
```
Comment (baywet) [inline src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs:45]:
```suggestion
                {s => s.StartsWith("x-") && !s.Equals("x-examples", StringComparison.OrdinalIgnoreCase), (o, p, n) => o.AddExtension(p, LoadExtension(p, n))}
```

File: src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs [hunk 2]
```diff
@@ -85,20 +91,58 @@ private static void ProcessProduces(MapNode mapNode, OpenApiResponse response, P
...
+            var examples = new Dictionary<string, OpenApiExample>();
+
+            foreach (var examplesNode in mapNode)
+            {
+                // Load the media type node as an OpenApiExample object
+                var example = new OpenApiExample();
+                var exampleNode = examplesNode.Value.CheckMapNode(examplesNode.Name);
+                foreach (var valueNode in exampleNode)
+                {
+                    switch (valueNode.Name)
```
Comment (baywet) [inline src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs:119]: do we need to normalize casing here?

File: src/Microsoft.OpenApi/Models/OpenApiResponse.cs [hunk 3]
```diff
@@ -231,6 +224,22 @@ public void SerializeAsV2WithoutReference(IOpenApiWriter writer)
                         writer.WriteEndObject();
                     }
 
+                    if (Content.Values.Any(m => m.Examples != null && m.Examples.Any()))
+                    {
+                        writer.WritePropertyName("x-examples");
```
Comment (baywet) [inline src/Microsoft.OpenApi/Models/OpenApiResponse.cs:229]: could we use a constant instead of defining it a couple of times all over the place?

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs (hunk 2 above)
Comment (MaggieKimani1) [reply src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs:119]: I don't think its necessary but I can add it just in case

Comment (baywet) [reply src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs:120]: where are those keys coming from? are they part of the schema or extension keys?

Comment (MaggieKimani1) [reply src/Microsoft.OpenApi.Readers/V2/OpenApiResponseDeserializer.cs]: Extension keys

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**